packages/labs/src/components/select/select.tsx

```diff
@@ -273,10 +273,10 @@ export class Select<T> extends React.Component<ISelectProps<T>, ISelectState<T>>
     private handleItemSelect = (item: T, event: React.SyntheticEvent<HTMLElement>) => {
         const { closeOnSelect = true } = this.props;
 
-        if (!closeOnSelect) {
-            event.stopPropagation();
-        } else {
+        if (closeOnSelect) {
             this.setState({ isOpen: false });
+        } else {
+            event.stopPropagation();
```

Comment:
an option I've considered here is calling `preventDefault()` here (downstream) and then in Popover close behavior (upstream), ignoring the event `if defaultPrevented`.